package_format/package_format.md

```diff
@@ -0,0 +1,45 @@
+Package format proposal
+=======================
+
+Requirements
+------------
+
+Package format should
+
+ * precisely describe a single 'app package' - how to install, run, restart, maintain working application
+   * be able to describe very custom apps with very custom requirements
+ * allow to generate user-friendly interface for installing apps
+   * provide information about how to configure an app
+   * provide information how to verify if configuration entered by user is valid
+   * provide information about needed dependencies
+   * provide sane default values
+ * allow to manualy configure app (with no checks and full responsibility on user)
+
+An idea
+-------
+
+Application package requires some dependencies and provides some dependencies for other packages. Single provided or required thing is called 'a resource'.
+
+Resource can be for example:
+
+ * string, int, bool or other simple datatype
+ * REST API available at given URL
+ * access to SQL DB (a tuple: URL, username, password)
+ * list of resources
+ * dictionary of resources (map string -> resource)
+
+Resources are typed. This provides additional verification if setup (set of installed apps) is valid.
+
+When installing an app user interface will generate form based on description which resources are needed. For example `string` resource will be displayed as text field and `REST API` will be displayed as dropdown list containing matching resources provided by other apps. After gathering configuration data it will be injected into commandline parameters or into configuration files of application.
+
+Beside of provided and required resources package contains other info. For example name, description, name of docker image, start command, reload command.
```

Comment:
`Apart from` instead of `Beside of`
`a package contains`
I'd use a comma before `for example` (and probably replace it with `e.g.`)
I think it's `Docker` (capital letter)